lib/ansible/modules/cloud/smartos/vmadm.py

```diff
@@ -277,10 +277,19 @@ def get_vm_uuid(module, alias):
         module.fail_json(
             msg='Could not retrieve UUID of {0}'.format(alias), exception=stderr)
 
-    try:
-        return json.loads(stdout)[0]['uuid']
-    except:
-        return None
+    # If no VM was found matching the given alias, we get back an empty array.
+    # That is not an error condition as we might be explicitly checking it's
+    # absence.
+    if stdout.strip() == '[]':
+      return None
+    else:
+      try:
+        stdout_json = json.loads(stdout)
+      except:
+        module.fail_json(msg='Invalid JSON returned by vmadm for uuid lookup of {0}'.format(alias))
```

Comment:
If getting the exception message, should put it in the message and use ```to_native(e)``` to retrieve it.  (to_native will avoid tracebacks if the string representation of e contains non-ascii).

if you want to use the exception field, it needs to contain the traceback which can be retrieved via the stdlib function: ```traceback.format_exc()```.